### PR TITLE
Fixed shooting while reloading

### PR DESCRIPTION
**Pull request by timmiedinnie**
- Fixed that you cannot shoot while reloading
- isolated some methods and made some imrpovements for readability and reusability

### CODE REVIEW

**Comment by timmiedinnie** [comment]
**Before**
before
**After:**
after

File: Assets/src/PlayerWeapons.cs [hunk 1]
```diff
@@ -68,6 +68,7 @@ public class PlayerWeapon {
   public Sprite Sprite => Weapon.Sprite;
   public Sprite InHandSprite => Weapon.InHandSprite;
   public AudioClip ShootSound => Weapon.ShootSound;
+  public bool isReloading = false;
```
Comment (12joan) [inline Assets/src/PlayerWeapons.cs:71]: We use PascalCase for class fields. It's probably not what most C# projects are using nowadays, but best to keep it consistent.

File: Assets/src/WeaponReload.cs [hunk 2]
```diff
@@ -8,51 +8,69 @@ public class WeaponReload : MonoBehaviour {
   public SoundController SoundController;
   public HUDBar ReloadBar;
   public MovementManager MovementManager;
-
-  IntervalTimer ReloadTimer;
+  private IntervalTimer ReloadTimer;
+  private const float movementSpeedReduction = 0.5f;
```
Comment (12joan) [inline Assets/src/WeaponReload.cs:12]: Same here with regard to PascalCase

File: Assets/src/WeaponReload.cs [hunk 3]
```diff
@@ -8,51 +8,69 @@ public class WeaponReload : MonoBehaviour {
...
 	void Start () {
-    ReloadTimer = new IntervalTimer() {
-      TimeClass = "PlayingTime"
-    };
-
-    // Reduce speed by half when reload is InProgress
+    InitializeReloadTimer();
+	}
+	
+  private void InitializeReloadTimer() {
```
Comment (12joan) [inline Assets/src/WeaponReload.cs:19]: There's probably not much point in putting this in its own function if it's the only thing called from Start. Can we revert this?

File: Assets/src/WeaponReload.cs [hunk 4]
```diff
@@ -8,51 +8,69 @@ public class WeaponReload : MonoBehaviour {
...
+    ReloadTimer = new IntervalTimer(){TimeClass = "PlayingTime"};
```
Comment (12joan) [inline Assets/src/WeaponReload.cs:20]:
```suggestion
    ReloadTimer = new IntervalTimer() {
      TimeClass = "PlayingTime"
    };

```

File: Assets/src/WeaponReload.cs [hunk 5]
```diff
@@ -8,51 +8,69 @@ public class WeaponReload : MonoBehaviour {
...
+    ReloadTimer.IfElapsed( CurrentWeapon().Reload );
+
+    if (UserTriggeredReload() && CanReload())
+      BeginReload();
+
+    UpdateReloadProgress();
+  }
+  private bool UserTriggeredReload() {
+    return WrappedInput.GetButtonDown("Reload") && !InProgress();
+  }
```
Comment (12joan) [inline Assets/src/WeaponReload.cs:58]: There seems to be some unintended duplication here? It's a little hard to make sense of what's going on with these changes.

File: Assets/src/WeaponReload.cs [hunk 6]
```diff
@@ -8,51 +8,69 @@ public class WeaponReload : MonoBehaviour {
...
+
+  private void StopReloading() {
+    CurrentWeapon().isReloading = false;
+    ReloadTimer.Stop();
+    SoundController.Stop();
+  }
```
Comment (12joan) [inline Assets/src/WeaponReload.cs:64]: Can you make sure to add an empty line between each method definition?

File: Assets/src/PlayerWeapons.cs (hunk 1 above)
Comment (12joan) [reply Assets/src/PlayerWeapons.cs:71]: Actually, on second thought, would it be possible to remove `isReloading` and use the existing `WeaponReload.InProgress()`  method instead?

**Review by 12joan — CHANGES_REQUESTED** [review]
I'm glad to see you were able to get Pop Pixie running.

In future, can you ask us on Discord before making any changes to the gameplay? As it happens, our team has talked about this particular change in the past, so I'm willing to give it a trial run to see how it plays.

I've left a few notes in the review comments. In addition, I noticed that `WeaponReload.cs` has an inconsistent mix of tabs and spaces that shows up on GitHub (an old issue from the code's early days). Would you mind fixing those since you're already editing the file? 🙂

**Pull request closed without merging**